### PR TITLE
task is now cancellable on "finish task" screen

### PR DESCRIPTION
**Pull request by deeayeen**

### CODE REVIEW

File: src/components/TaskList/UserFinish.js [hunk 1]
```diff
@@ -18,9 +20,28 @@ export default function UserFinish(props) {
...
+    };
+
+    const handleCancelTask = async () => {
+        try {
+            await wrapWithLoading(async () => {
+                const id = task._id;
+                const newTask = {
+                    ...task,
+                    cancelMessage: message,
+                    status: "cancelled"
```
Comment (amitaiblevy) [inline src/components/TaskList/UserFinish.js:40]: Discuss

File: src/components/TaskList/UserFinish.js [hunk 2]
```diff
@@ -65,15 +83,37 @@ export default function UserFinish(props) {
...
-                        לא
+                        ביטול הדרישה
                     </Button>
                 </div>
+                {cancelTask && (
+                    <div style={{ marginTop: "32px", display: "flex", alignItems: "center" }}>
+                        <TextField
+                            value={message || ""}
+                            variant="outlined"
+                            onChange={e => setMessage(e.target.value)}
```
Comment (amitaiblevy) [inline src/components/TaskList/UserFinish.js:99]: Use getOnTextChange from utils

File: src/components/TaskList/UserFinish.js [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+import { Button, Typography, TextField } from "@material-ui/core";
 import useGlobalState from "../../state";
 import { updateTask } from "../../utils/api";
 
 export default function UserFinish(props) {
     const { wrapWithLoading } = useGlobalState();
-    const { task, onConfirm, onCancel = (() => { }), onClose } = props;
+    const { task, onConfirm, onCancel = () => {}, onClose } = props;
+    const [cancelTask, setCancelTask] = useState(false);
+    const [message, setMessage] = useState("");
```
Comment (amitaiblevy) [inline src/components/TaskList/UserFinish.js:11]: default null. see below comment

File: src/components/TaskList/UserFinish.js (hunk 2 above)
Comment (deeayeen) [reply src/components/TaskList/UserFinish.js]: done

File: src/components/TaskList/UserFinish.js (hunk 3 above)
Comment (deeayeen) [reply src/components/TaskList/UserFinish.js]: done

File: src/components/TaskList/UserFinish.js [hunk 4]
```diff
@@ -18,9 +21,28 @@ export default function UserFinish(props) {
...
+        }
+    };
+
+    const handleCancelTask = async () => {
+        try {
+            await wrapWithLoading(async () => {
+                const id = task._id;
+                const newTask = {
+                    ...task,
+                    cancelMessage: message,
```
Comment (amitaiblevy) [inline src/components/TaskList/UserFinish.js:40]: Add cancelledStatusDate field

**Pull request closed without merging**